Kitodo/src/main/java/org/kitodo/production/services/dataformat/MetsService.java

```diff
@@ -85,6 +95,25 @@ public Workpiece loadWorkpiece(URI uri) throws IOException {
         }
     }
 
+    /**
+     * Create and return Workpiece from given Document 'document'.
+     *
+     * @param document Document from which a Workpiece is created
+     * @return Workpiece created from given Document
+     * @throws TransformerException
+     *          thrown when XML transformation fails
+     * @throws IOException
+     *          thrown when unable to read inputStream
+     */
+    public Workpiece loadWorkpiece(Document document) throws TransformerException, IOException {
```

Comment:
In this method a lot of stream objects are created / opened but not explicit closed. This can result in open stream handles which are only will closed if the application will be stopped. Maybe use of try-with-resources statement could improve automatic closing of stream objects.

Reply:
I commited this method by accident, it wasn't supposed to be included in this Pull Request, therefore I removed it for now.